Source/Cypher/app/src/main/java/com/semaphore_soft/apps/cypher/ConnectionLobbyActivity.java

```diff
@@ -67,6 +90,41 @@ public void onCreate(Bundle savedInstanceState)
         {
             playerID = 0;
 
+            String ip = "";
+            try
+            {
+                for (Enumeration<NetworkInterface> en = NetworkInterface.getNetworkInterfaces(); en.hasMoreElements();)
+                {
+                    NetworkInterface ni = en.nextElement();
+                    for (Enumeration<InetAddress> addresses = ni.getInetAddresses(); addresses.hasMoreElements();)
+                    {
+                        InetAddress inetAddress = addresses.nextElement();
+                        // Limit IP addresses shown to IPv4
+                        if (!inetAddress.isLoopbackAddress() && inetAddress instanceof Inet4Address)
+                        {
+                            ip = inetAddress.getHostAddress();
+                            Log.i("Lobby", ip);
+                        }
+                    }
+                }
+            }
+            catch (SocketException ex)
+            {
+                Log.e("Lobby", ex.toString());
+            }
+            
+            mServiceIntent = new Intent(this, NetworkingService.class);
+            mServiceIntent.setData(Uri.parse(NetworkingService.SETUP_SERVER));
+            startService(mServiceIntent);
+
+            mServiceIntent.setData(Uri.parse(NetworkingService.WRITE_TO_CLIENT));
+            mServiceIntent.putExtra("message", "Hello, World!");
+            mServiceIntent.putExtra("index", 0);
+            startService(mServiceIntent);
+
+            TextView ipAddress = (TextView) findViewById(R.id.ip_address);
+            ipAddress.setText("Your IP Address is: " + ip);
+
```

Comment:
Since the client and the server shouldn't be running on the same device whether or not the host has started accepting is going to be beyond the client's control anyway, but I think it would also make sense to start accepting as soon as we've got the host's name or start trying to connect as soon as we've got the client's name and host IP, on the callback from that fragment in the main activity. But I guess I could see it either way. Probably doesn't make much functional difference, just want to make sure we're doing it when and where it makes the most sense.